vega/base.py

```diff
@@ -18,8 +19,8 @@ class VegaBase(object):
 
     def __init__(self, spec, data=None):
         """Initialize the visualization object."""
-        spec = self._prepare_spec(spec, data)
-        self.spec = utils.update(spec, self.DEFAULTS, overwrite=False)
+        spec = self._prepare_spec(copy.deepcopy(spec), data)
+        self.spec = utils.update(spec, self.DEFAULTS)
```

Comment:
I noticed there's a similar ``deepcopy`` at another place in the utils. We might fix that as well.